test/e2e/validatingwebhook/kubeclient.go

```diff
@@ -0,0 +1,172 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package validatingwebhook
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/mitchellh/go-homedir"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	namespace = "default"
+)
+
+// KubernetesClient will connect to local k8s cluster,
+// and help perform resource operation
+type KubernetesClient struct {
+	client *kubernetes.Clientset
+}
+
+// NewKubernetesClient creates a new Kubernetes client
+func NewKubernetesClient() (*KubernetesClient, error) {
+	kubernetesClient := new(KubernetesClient)
+	var err error
+	kubernetesClient.client, err = kubernetesClient.getK8sClient()
+	if err != nil {
+		return nil, err
+	}
+	return kubernetesClient, nil
+}
+
+// getK8sClient creates a kubernetes clientset with default config path
+func (k *KubernetesClient) getK8sClient() (*kubernetes.Clientset, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return nil, fmt.Errorf("home directory not found, error: %s", err.Error())
+	}
+
+	configPath := filepath.Join(home, ".kube", "config")
```

Comment:
how are we sure that the kubeconfig at this path is always setup with the context that deal with our desired k8s cluster?

Reply:
it'd be better if we could bring KUBECONFIG env variable to use.

Reply:
In our test environment, we are creating a kubernetes cluster using `kind create cluster`, it will ensure that a config file is present in the default kubeconfig path.

Reply:
had a call with @dev-gaur to verify that existing test implementation will work fine.